### PR TITLE
Initialize callback functions into libRmath-julia

### PR DESCRIPTION
**Pull request by simonbyrne**
see https://github.com/JuliaLang/Rmath-julia/pull/13

### CODE REVIEW

File: src/Rmath.jl [hunk 1]
```diff
@@ -42,6 +42,16 @@ end
     export dwilcox,pwilcox,qwilcox,rwilcox # Wilcox's Rank Sum statistic (m, n)
     export ptukey, qtukey              # Studentized Range Distribution - p and q only
 
+function __init__()
+
+    unsafe_store!(cglobal((:unif_rand_ptr,libRmath),Ptr{Void}),
```
Comment (ararslan) [inline src/Rmath.jl]: Looks like `cglobal` can't find `unif_rand_ptr` in libRmath.

Comment (simonbyrne) [inline src/Rmath.jl]: yes, this requires the corresponding change to Rmath-julia (I should have explained that in the PR comment)

Comment (ararslan) [inline src/Rmath.jl]: Oh, okay. Don't mind me then. 😄

**Comment by tkelman** [comment]
so https://github.com/JuliaStats/Rmath.jl/pull/18 didn't fix this?

**Comment by simonbyrne** [comment]
Yes, you're right. I had forgotten to Pkg.update. This did seem to fix it.

This solution does get rid of the parallel streams which we're using at the moment, so still might be worth trying out.

**Comment by tkelman** [comment]
I'm not sure if it would have auto-redone Pkg.build to recompile the library, so that might have also been stale

File: src/Rmath.jl [hunk 2]
```diff
@@ -42,6 +42,16 @@ end
     export dwilcox,pwilcox,qwilcox,rwilcox # Wilcox's Rank Sum statistic (m, n)
     export ptukey, qtukey              # Studentized Range Distribution - p and q only
 
+function __init__()
+
```
Comment (nalimilan) [inline src/Rmath.jl]: No line breaks at the top of function bodies.

**Comment by simonbyrne** [comment]
I still think this is a worthwhile change to make, as it will allow us to use alternatives to DSFMT, and make the overall interface much clearer.

**Comment by tkelman** [comment]
Sure, if it's worth updating the library for. Should add a validation hook to check that these functions are present. That would be a pretty good screen.

**Comment by simonbyrne** [comment]
Okay, I'll look into it. I might also try to update the R sources while I'm at it, I think it's been a while.

**Comment by simonbyrne** [comment]
Bump?

**Comment by tkelman** [comment]
sorry, where were we on this?

**Comment by simonbyrne** [comment]
I think what needs to be done:
1. Merge https://github.com/JuliaLang/Rmath-julia/pull/14
2. Tag a new version of Rmath-julia and create a downloadable windows binary
3. Bump the version requirement in this PR
4. Merge this.

**Comment by tkelman** [comment]
> Insert julia v0.6 upper bound in METADATA for old versions of Rmath.jl

What's the reason for that bit?

**Comment by simonbyrne** [comment]
@tkelman Can you create a new windows binary for https://github.com/JuliaLang/Rmath-julia/releases/tag/v0.2.0, that will get step 2 done.

Item 6 will break random number generator used by the current Rmath binary.

**Comment by ararslan** [comment]
Regarding step 2, we can make macOS and Linux binaries in a similar manner as https://github.com/ararslan/fftw-builder.

**Comment by simonbyrne** [comment]
That would be great!

**Comment by tkelman** [comment]
If I can remember how the library used to integrate in Base's build system to make sure it gets built correctly. I'll probably get it written in an appveyor.yml on Rmath-julia, since we now have Base's appveyor using cygwin-cross and it wasn't too bad to get set up. Who has space on their appveyor account for it to run?

**Comment by simonbyrne** [comment]
It should be simpler now, as we don't need the dSFMT headers. I think I have appveyor space

**Comment by simonbyrne** [comment]
Okay, it should be enabled now.

**Comment by simonbyrne** [comment]
This should be good to go once CI passes.

File: src/Rmath.jl (hunk 2 above)
Comment (tkelman) [inline src/Rmath.jl]: not addressed

Comment (simonbyrne) [inline src/Rmath.jl]: now fixed

**Pull request merged**